src/shogun/distributions/HMM.h

```diff
@@ -368,6 +400,19 @@ class Model
  * Several functions for tasks such as training,reading/writing models, reading observations,
  * calculation of derivatives are supplied.
  */
+
+/** @brief CHMM
+ * This class implements forward - backward algorithm.
+ * Forward algorithm computes joint probability distribution P(z_k,x_1 to k) for all k = 1 to n. (Method - forward_comp( ))
+ * Backward algorithm computes P(x_k+1 to n|z_k) for all k = 1 to n. (Method - backward_comp( ))
+ * baum_welch_viterbi_train(BaumWelchViterbiType type) is used to run BaumWelch or Viterbi training.
+ * Various algorithms implemented to train fully connected HMM and defined transitions are :
+ * baum-welch-algorithm and viterbi training. (Methods - estimate_model_baum_welch(CHMM* train), estimate_model_viterbi(CHMM* train) etc)
+ * To estimate linear model from observations linear_train method is implemented.
+ * Also to satisfy stochasticity normalize method is implemented.
+ * Best path is calculated using viterbi algorithm in method best_path( ).
```

Comment:
please make all references in such way that doxygen can create links, you can create the pages locally via running 
```make doc```